concent_api/core/payments/bankster.py

```diff
@@ -246,14 +247,15 @@ def finalize_payment(deposit_claim: DepositClaim) -> Optional[str]:
         elif deposit_claim.concent_use_case == ConcentUseCase.ADDITIONAL_VERIFICATION:
             subtask = Subtask.objects.filter(subtask_id=deposit_claim.subtask_id).first()  # pylint: disable=no-member
             if subtask is not None:
-                if subtask.requestor.public_key == deposit_claim.payer_deposit_account.client.public_key:
+                task_to_compute = deserialize_message(subtask.task_to_compute.data.tobytes())
+                if task_to_compute.requestor_ethereum_address == deposit_claim.payer_deposit_account.ethereum_address:
```

Comment:
Could you explain what this pull request is actually fixing? Neither the pull request nor the commit description gives any clue and there's no issue for it.

The condition should be fine no matter whether you compare the public key, the ethereum address or even just the IDs of the client object. They should be equivalent here.

Reply:
I agree, it should work the same for both the same for the cases which you have described, but my guess (I wasn't able to reproduce exception described on slack) is that it might be a problem with test, that we are using same `subtask_id` with different clients. And as we have `deposit_claim.payer_deposit_account.ethereum_address` unique, it might fix the original issue.

Reply:
Please first test and make sure that it really fixes anything.

I really don't like magical changes like that. Especially if there're random tweaks just to safisfy the tests. If our tests are reusing IDs, that's what we should fix - please submit an issue.

If the problem is not reproducible, I'd prefer not to merge this change.